Lib/sqlite3/test/userfunctions.py

```diff
@@ -275,6 +276,28 @@ def CheckAnyArguments(self):
         val = cur.fetchone()[0]
         self.assertEqual(val, 2)
 
+    def CheckFuncDeterministic(self):
```

Comment:
I think this test can be split into three tests:

1. `deterministic=False`
2. `deterministic=True` if SQLite is older than 3.8.3
3. `deterministic=True` if SQLite supports the feature